lib/TPP/Transforms/LinalgConvertCompareSelectToMaximumfPass.cpp

```diff
@@ -0,0 +1,81 @@
+//===-LinalgConvertCompareSelectToMaximumfPass.cpp ---------------*- C++-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file lowers Compare select generic to maximumf generic
+//
+//===----------------------------------------------------------------------===//
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+namespace mlir {
+namespace tpp {
+#define GEN_PASS_DEF_LINALGCONVERTCOMPARESELECTTOMAXIMUMFPASS
+#include "TPP/Passes.h.inc"
+} // namespace tpp
+} // namespace mlir
+
+using namespace mlir;
+using namespace mlir::linalg;
+
+namespace mlir {
+namespace tpp {
+
+struct LinalgConvertCompareSelectToMaximumf
+    : public OpRewritePattern<linalg::GenericOp> {
+  using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::GenericOp op,
+                                PatternRewriter &rewriter) const override {
+
+    if (op.getBody()->getOperations().size() != 3)
+      return failure();
+    auto cmpf = dyn_cast<arith::CmpFOp>(&op.getBody()->getOperations().front());
+    if (!cmpf)
+      return failure();
+    auto select = dyn_cast<arith::SelectOp>(
+        std::next(op.getBody()->getOperations().begin(), 1));
+    if (!select)
+      return failure();
```

Comment:
I'd also add a check to make sure the `cmp` is with `zero` and the `select` is between the result of `cmp` and `zero`.

Two simple tests not matching:
* comparison / select with non-zero
* comparison / select with zero but comparison from an external (captured) value, not the result of `cmp`